modules/mapping-utils/src/resolveSets.js

```diff
@@ -149,48 +175,193 @@ export const deleteSaveSets = ({ callback }) => async (
   return esResponse.deleted;
 };
 
-export const renameSaveSetTag = ({ callback }) => async (
+export const updateSet = ({ types, callback }) => async (
   obj,
-  { setId, tag, userId },
+  { source, subAction, target, userId, data },
   { es },
 ) => {
-  const esResponse = await es.updateByQuery({
-    index: CONSTANTS.ES_ARRANGER_SET_INDEX,
-    type: CONSTANTS.ES_ARRANGER_SET_TYPE,
-    refresh: true,
-    body: {
-      script: {
-        lang: 'painless',
-        source: `ctx._source['tag'] = '${tag}'`,
-      },
-      query: {
-        bool: {
-          filter: {
-            term: { userId: userId },
+  const { sourceType } = source;
+
+  switch (subAction) {
```

Comment:
this switch case is way too big. pls split it in logic blocks, inner function and then a clean switch statement.

Reply:
Good point